addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/OpenApiSpider.java

```diff
@@ -64,11 +64,15 @@ public boolean parseResource(HttpMessage message, Source source, int depth) {
     @Override
     public boolean canParseResource(HttpMessage message, String path, boolean wasAlreadyConsumed) {
         try {
-            String contentType = message.getResponseHeader().getHeader(HttpHeader.CONTENT_TYPE);
-            if (contentType.toLowerCase().startsWith("text")
-                    && message.getResponseBody().toString().toLowerCase().startsWith("swagger:")) {
+            String contentType =
+                    message.getResponseHeader().getHeader(HttpHeader.CONTENT_TYPE).toLowerCase();
+            String responseBodyStart =
+                    message.getResponseBody().toString().substring(0, 8).toLowerCase();
+            if (contentType.startsWith("application/vnd.oai.openapi")) {
                 return true;
-            } else if (contentType.toLowerCase().startsWith("application")) {
+            } else if ((contentType.contains("json") || contentType.contains("yaml"))
+                    && (responseBodyStart.equals("swagger:")
```

Comment:
These checks might not work properly (e.g. if the OpenAPI has comments or when JSON which starts with `{`), some examples in https://github.com/OAI/OpenAPI-Specification/tree/master/examples